interpreter/cling/lib/Interpreter/DynamicLibraryManager.cpp

```diff
@@ -152,12 +187,36 @@ namespace cling {
         return std::string();
     }
 
-    std::string foundName = lookupLibMaybeAddExt(libStem);
-    if (foundName.empty() && !libStem.startswith("lib")) {
-      // try with "lib" prefix:
-      foundName = lookupLibMaybeAddExt("lib" + libStem.str());
+    // Handle substitutions
+    std::string result;
+    llvm::StringRef mainExecutable =
+        llvm::sys::fs::getMainExecutable(nullptr, nullptr);
+    if (libStem.startswith_lower("@rpath/")) {
+      llvm::SmallVector<llvm::StringRef, 2> RPaths;
+      SplitPaths(RPATH, RPaths, utils::kPruneNonExistant, platform::kEnvDelim,
+                 false);
+      for (auto& P : RPaths) {
+        subst(libStem, "@rpath/", P.str(), result);
+        subst(result, "@executable_path/", mainExecutable, result);
```

Comment:
IIUC `@executable_path` expects the *directory* of the binary location, while `getMainExecutable()` returns the full path to the image file. I.e. there's "get the directory of `mainExecutable`" missing somewhere?

Reply:
Thanks fixed!